### PR TITLE
Add SocketPath/PciID to Interface struct 

### PR DESCRIPTION
**Pull request by MikeZappa87**
This is to resolve: #1050

### CODE REVIEW

**Comment by squeed** [comment]
Looks good; we need to add this to the spec, though.

**Comment by MikeZappa87** [comment]
> Looks good; we need to add this to the spec, though.

I can start on that right now.

File: pkg/types/100/types_test.go [hunk 1]
```diff
@@ -106,7 +107,9 @@ var _ = Describe("Current types operations", func() {
         {
             "name": "eth0",
             "mac": "00:11:22:33:44:55",
-            "sandbox": "/proc/3553/ns/net"
+            "sandbox": "/proc/3553/ns/net",
+			"pci_id": 	 "8086:9a01",
```
Comment (squeed) [inline pkg/types/100/types_test.go:111]: nit: spacing here seems off?

**Review by tamilmani1989 — APPROVED** [review]
lgtm

File: SPEC.md [hunk 2]
```diff
@@ -565,9 +565,11 @@ Plugins must output a JSON object with the following keys upon a successful `ADD
...
 - `cniVersion`: The same version supplied on input - the string "1.1.0"
 - `interfaces`: An array of all interfaces created by the attachment, including any host-level interfaces:
-    - `name`: The name of the interface.
-    - `mac`: The hardware address of the interface (if applicable).
-    - `sandbox`: The isolation domain reference (e.g. path to network namespace) for the interface, or empty if on the host. For interfaces created inside the container, this should be the value passed via `CNI_NETNS`.
+    - `name` (string): The name of the interface.
+    - `mac` (string): The hardware address of the interface (if applicable).
+    - `sandbox` (string): The isolation domain reference (e.g. path to network namespace) for the interface, or empty if on the host. For interfaces created inside the container, this should be the value passed via `CNI_NETNS`.
+    - `socket_path` (string): The path of the socket that can be passed down to OCI runtimes.
+    - `pci_id` (string): The combination of vendor id and device id to uniquely identify a device.
```
Comment (s1061123) [inline SPEC.md:572]: Could you change go convention, as camel case? (i.e. socketPath / pciID)

**Review by s1061123 — CHANGES_REQUESTED** [review]

File: SPEC.md [hunk 3]
```diff
@@ -565,9 +565,11 @@ Plugins must output a JSON object with the following keys upon a successful `ADD
 
 - `cniVersion`: The same version supplied on input - the string "1.1.0"
 - `interfaces`: An array of all interfaces created by the attachment, including any host-level interfaces:
-    - `name`: The name of the interface.
-    - `mac`: The hardware address of the interface (if applicable).
-    - `sandbox`: The isolation domain reference (e.g. path to network namespace) for the interface, or empty if on the host. For interfaces created inside the container, this should be the value passed via `CNI_NETNS`.
+    - `name` (string): The name of the interface.
+    - `mac` (string): The hardware address of the interface (if applicable).
+    - `sandbox` (string): The isolation domain reference (e.g. path to network namespace) for the interface, or empty if on the host. For interfaces created inside the container, this should be the value passed via `CNI_NETNS`.
+    - `socket_path` (string): The path of the socket that can be passed down to OCI runtimes.
```
Comment (squeed) [inline SPEC.md:571]:
```suggestion
    - `socket_path` (string, optional): An absolute path to a socket file corresponding to this interface, if applicable.
```

File: SPEC.md (hunk 2 above)
Comment (squeed) [inline SPEC.md:572]:
```suggestion
    - `pciID` (string, optional): The platform-specific identifier of the PCI device corresponding to this interface, if applicable.
```

**Review by s1061123 — APPROVED** [review]

**Pull request merged**